astropy/wcs/tests/test_wcs.py

```diff
@@ -989,6 +993,108 @@ def test_sip_tpv_agreement():
         )
 
 
+def test_tpv_ctype_sip():
+    sip_header = fits.Header.fromstring(
+        get_pkg_data_contents(os.path.join("data", "siponly.hdr"), encoding="binary")
+    )
+    tpv_header = fits.Header.fromstring(
+        get_pkg_data_contents(os.path.join("data", "tpvonly.hdr"), encoding="binary")
+    )
+    sip_header.update(tpv_header)
+    sip_header["CTYPE1"] = "RA---TAN-SIP"
+    sip_header["CTYPE2"] = "DEC--TAN-SIP"
+
+    with pytest.warns(
+        wcs.FITSFixedWarning,
+        match="Removed redundant SCAMP distortion parameters "
+        "because SIP parameters are also present",
+    ):
+        w_sip = wcs.WCS(sip_header)
+
+    assert w_sip.sip is not None
+
+
+def test_tpv_ctype_tpv():
+    sip_header = fits.Header.fromstring(
+        get_pkg_data_contents(os.path.join("data", "siponly.hdr"), encoding="binary")
+    )
+    tpv_header = fits.Header.fromstring(
+        get_pkg_data_contents(os.path.join("data", "tpvonly.hdr"), encoding="binary")
+    )
+    sip_header.update(tpv_header)
+    sip_header["CTYPE1"] = "RA---TPV"
+    sip_header["CTYPE2"] = "DEC--TPV"
+
+    with pytest.warns(
+        wcs.FITSFixedWarning,
+        match="Removed redundant SIP distortion parameters "
+        "because CTYPE explicitly specifies TPV distortions",
+    ):
+        w_sip = wcs.WCS(sip_header)
+
+    assert w_sip.sip is None
+
+
+def test_tpv_ctype_tan():
+    sip_header = fits.Header.fromstring(
+        get_pkg_data_contents(os.path.join("data", "siponly.hdr"), encoding="binary")
+    )
+    tpv_header = fits.Header.fromstring(
+        get_pkg_data_contents(os.path.join("data", "tpvonly.hdr"), encoding="binary")
+    )
+    sip_header.update(tpv_header)
+    sip_header["CTYPE1"] = "RA---TAN"
+    sip_header["CTYPE2"] = "DEC--TAN"
+
+    with pytest.warns(
+        wcs.FITSFixedWarning,
+        match="Removed redundant SIP distortion parameters "
+        "because SCAMP' PV distortions are also present",
+    ):
+        w_sip = wcs.WCS(sip_header)
+
+    assert w_sip.sip is None
+
+
+def test_car_sip_with_pv():
+    # https://github.com/astropy/astropy/issues/14255
+    header_dict = {
+        "SIMPLE": True,
+        "BITPIX": -32,
+        "NAXIS": 2,
+        "NAXIS1": 1024,
+        "NAXIS2": 1024,
+        "CRPIX1": 512.0,
+        "CRPIX2": 512.0,
+        "CDELT1": 0.01,
+        "CDELT2": 0.01,
+        "CRVAL1": 120.0,
+        "CRVAL2": 29.0,
+        "CTYPE1": "RA---CAR-SIP",
+        "CTYPE2": "DEC--CAR-SIP",
+        "PV1_1": 120.0,
+        "PV1_2": 29.0,
+        "PV1_0": 1.0,
+        "A_ORDER": 2,
+        "A_2_0": 5.0e-4,
+        "B_ORDER": 2,
+        "B_2_0": 5.0e-4,
+    }
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        w = wcs.WCS(header_dict)
```

Comment:
Unless your warning has been globally ignored in setup.cfg , this is indeed unnecessary. Making it explicit is just more unnecessary code to maintain. And if we really need to explicitly check like this, our whole test suit is in trouble.

```suggestion
    w = wcs.WCS(header_dict)
```

Reply:
Agreed with @pllim really - best to just keep it simple (but add a comment)